Derpy/Palette/Service.cs

```diff
@@ -0,0 +1,34 @@
+using System.Net.Http;
+using System.Threading.Tasks;
+using Derpy.Result;
+
+namespace Derpy.Palette
+{
+    public class Service
+    {
+        public const string RANDOM_COLOUR_PALETTE_URL = "https://www.colourpod.com/random";
+        private readonly HttpClient _httpClient;
+
+        public Service(HttpMessageHandler handler)
+        {
+            _httpClient = new HttpClient(handler);
```

Comment:
I read in the documentation that you should have a single `HttpClient` instance for your whole program, wouldn't it make sense to put the `HttpClient` instance itself in the `ServiceCollection`?

Reply:
Ah, that's a fair call. The reason I did it this way was so I could mock out the `HttpClient` behaviour for unit testing via the internal `SendAsync` method of the `HttpResponseMessage` - very messy. A (much) nicer thing to do would be to write a wrapper class around `HttpClient` that can be mocked separately, as well as added as a singleton to the `ServiceCollection`.